src/config.rs

```diff
@@ -509,10 +523,16 @@ pub fn get_config_dir() -> Result<(PathBuf, PathSource)> {
 ///
 /// Note that this function does not verify whether the file at that location
 /// exists, or is a file.
-pub fn get_config_path() -> Result<(PathBuf, PathSource)> {
-    let (config_dir, source) = get_config_dir()?;
-    let config_file_path = config_dir.join(CONFIG_FILE_NAME);
-    Ok((config_file_path, source))
+pub fn get_config_path(path: Option<&Path>) -> Result<(PathBuf, PathSource)> {
```

Comment:
I think we should keep this function unchanged, except for a rename to `get_default_config_path`; the CLI override logic is not really a concern for the config.rs file